src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -514,6 +514,7 @@ public PossiblyConditionalState Clone()
                                         var outputSlot = makeDagTempSlot(type, output);
                                         Debug.Assert(outputSlot > 0);
                                         addToTempMap(output, outputSlot, type.Type);
+                                        this.State[outputSlot] = NullableFlowState.NotNull; // Slice value is assumed to be never null
```

Comment:
❓ Should we also strip the annotation on the return type for semantic model?

Reply:
The `var` will definitely have a `?`, because it always does. Theoretically, though, it should say that `variable` is considered not null here.